### PR TITLE
Completed challenge 82, with test

### PR DESCRIPTION
**Pull request by gaebar**

### CODE REVIEW

**Review by rrees — COMMENTED** [review]
Your test cases need to include 0, negative numbers and non-numbers

File: exercises/more_string_manipulation/challenge_082.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    length_poem = len(poem)
+    if length_poem < start:
+        raise IndexError("list index out of range")
+
+    if isinstance(start, int) and isinstance(end, int):
+        if start > 0 and end >= start:
+            user_line_choose = poem[start:end]
+            return user_line_choose
+        else:
+            raise TypeError("Input indexes are out of range")
```
Comment (rrees) [inline exercises/more_string_manipulation/challenge_082.py]: Why is this a `TypeError` when the first bound-check is an `IndexError`? This seems inconsistent

File: tests/more_string_manipulation/test_challenge_082.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+@pytest.mark.parametrize("poem, start, end, expected", testdata)
+def test_trim_characters(poem, start, end, expected):
+ 
+    trimmed_poem = challenge_082.trim_characters(poem, start, end)
+
+    assert trimmed_poem == expected
+
+@pytest.mark.parametrize("poem, start, end, expected", testdata_fails)
```
Comment (rrees) [inline tests/more_string_manipulation/test_challenge_082.py]: I would separate the different kind of failure cases into different test cases. Mixing together different failure cases makes it difficult to expand test suites later or expand on different kind of failure cases.

This is also where doing TDD versus test after makes creating appropriate tests easier

**Review by rrees — APPROVED** [review]

File: exercises/more_string_manipulation/challenge_082.py (hunk 1 above)
Comment (gaebar) [reply exercises/more_string_manipulation/challenge_082.py]: amended exceptions raised. Thanks for noticing!

File: tests/more_string_manipulation/test_challenge_082.py (hunk 2 above)
Comment (gaebar) [reply tests/more_string_manipulation/test_challenge_082.py]: Separated into TypeError and IndexError test cases

**Pull request merged**